src/lib/y2partitioner/format_mount_options.rb

```diff
@@ -0,0 +1,293 @@
+# encoding: utf-8
+
+# Copyright (c) [2017] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "y2storage"
+
+module Y2Storage
+  # FIXME: Monkey patch of Y2Storage::PartitionId it should moved to
+  # yast2-storage-ng
+  class PartitionId
+    include Yast::I18n
+    extend Yast::I18n
+
+    TRANSLATIONS = {
+      dos12:              N_("DOS12"),
+      dos16:              N_("DOS16"),
+      dos32:              N_("DOS32"),
+      swap:               N_("Linux Swap"),
+      linux:              N_("Lnux"),
+      lvm:                N_("Linux LVM"),
+      raid:               N_("Linux RAID"),
+      esp:                N_("EFI System Partition"),
+      bios_boot:          N_("BIOS Boot Partition"),
+      prep:               N_("PReP Boot Partition"),
+      ntfs:               N_("NTFS"),
+      extended:           N_("Extended"),
+      windows_basic_data: N_("Windows Data Partition"),
+      microsoft_reserved: N_("Microsoft Reserved Partition"),
+      diag:               N_("Diagnostics Partition"),
+      unknown:            N_("Unknown")
+    }.freeze
+    private_constant :TRANSLATIONS
+
+    def to_human_string
+      textdomain "storage"
+
+      string = TRANSLATIONS[to_sym] or raise "Unhandled Partition ID '#{inspect}'"
+
+      _(string)
+    end
+
+    def formattable?
+      !%i(lvm raid esp prep bios_boot unknown).include?(to_sym)
+    end
+  end
+
+  module Filesystems
+    # FIXME: Temporal Monkey patching of fstab options per filesystem type. It
+    # could/should be moved to yast2-storage-ng.
+    class Type
```

Comment:
In general it started to miss a specific class for each type. Maybe make sense to create one per filesystem in the `Y2Partitioner` namespace instead of directly under `Y2Storage` one by now.